### DISCUSSION TITLE
Sales and Discounts

### DISCUSSION THREAD

**Discussion by meska** [saleor/saleor#7362] [Q&A]
Hello, is there a specific reason why sales are not publicly readable but protected by DiscountPermissions.MANAGE_DISCOUNTS ?
How can I query a list of sales from the shop ?

**Comment by TurkaljGoran** [comment]
Can we get an answer for this please?

Currently we're trying to fetch products on sale to display them on a specific page and it would be great if sales could be queried without needing MANAGE_DISCOUNTS permission OR if products could be filtered by 'onSale' property or something like that. 

We'll probably end up filling metadata field on all products with a key 'onSale': true, but this will require looping over all products, checking if they are on sale, then adding the metadata key to them, then filtering on the frontend by that metadata key, but it would be nice if this functionality was implemented natively.

Cheers!

**Reply by fmravic96** [reply]
Hello,

We are working on this also. Can we get an answer soon please?

**Reply by karolkielecki** [reply]
Let me ask the team if we can do something about it.

**Reply by maarcingebala** [reply]
There is no filter like that currently, there is only `onSale` flag in the product pricing:
```
{
  products(first: 5, channel: "default-channel") {
    edges {
      node {
        id
        pricing{
          onSale
        }
      }
    }
  }
}
```

One quick workaround I see here is to create a collection and put all products on sale to this collection, then filter products by this collection ID:
```
products(first: 5, channel: "default-channel", filter: {collections: ["collection-id-1"]}) { ... }
```

**Comment by timuric** [comment]
Related issue https://github.com/saleor/saleor/issues/9478
Filter enhancements are on the roadmap

**Comment by inf3cti0n95** [comment]
any updates on this?

**Comment by dergachevm** [comment]
Really no "native" way to request only those products that are on discount?
